### DISCUSSION TITLE
Unable to expose another schema (Hosted version of supabase)

### DISCUSSION THREAD

**Discussion by PeterReiter** [supabase/supabase#7642] [Questions] [answered]
Dear community, my name is Peter Reiter, I'm a new user of supabase.

What ist the recommended (and working) approach to expose a 'custom' schema instead of the default 'public'.
What I have tried so far are the following steps:

1. Create a new schema and grant the same privileges as available for the 'public' schema.
2. Add the schema to the list of schemas in the API settings field.
3. Create a simple 'test' table with the same privileges as a table created in the 'public' schema.

With best regards,
Peter Reiter

**Comment by GaryAustin1** [comment, marked as answer]
You don’t actually say what your issue is… those steps seem correct. 

if you mean how to access the schema from the client, here is the documented way was a parameter at init of Supabase-js … https://supabase.com/docs/reference/javascript/initializing#with-additional-parameters

Also here I suggest two other more useful ways that seem to work, and SB employees discuss some future stuff. https://github.com/supabase/postgrest-js/issues/280#issuecomment-1152758870

**Reply by PeterReiter** [reply]
> You don’t actually say what your issue is… those steps seem correct.
> 
> if you mean how to access the schema from the client, here is the documented way was a parameter at init of Supabase-js … https://supabase.com/docs/reference/javascript/initializing#with-additional-parameters
> 
> Also here I suggest two other more useful ways that seem to work, and SB employees discuss some future stuff. [supabase/postgrest-js#280 (comment)](https://github.com/supabase/postgrest-js/issues/280#issuecomment-1152758870)

The issue is, that the table defined in the new schema, ist not listed in the UI, see atached image (test2 is a table in the public schema).

![image](https://user-images.githubusercontent.com/5524586/178257071-b5fbc9ba-79b1-49ee-9563-4ecf3a44c54e.png)

**Comment by GaryAustin1** [comment]
Ah.both the database tab and the API tab do not have option to see other schemas. 
if you go to the table Ui though you should have the the option to select the schema:
![36DB0FE9-7EF7-4E76-93FD-B2A4CFD80D27](https://user-images.githubusercontent.com/54564956/178262036-400966df-addc-4088-95cb-7f8f3da8a81c.jpeg)

and the table should be reachable from the clients if your grants were done correctly.

**Reply by PeterReiter** [reply]
The table in the "other" schema is reachable using an js-client, do you know how this can be achieved using the REST API?

**Reply by GaryAustin1** [reply]
This piece of code shows the headers involved:
https://github.com/supabase/postgrest-js/blob/a68d7dbd5f61046f21297fb78e6407c510bc1df4/src/lib/types.ts#L99

**Reply by PeterReiter** [reply]
Thanks, for you help. Everzthin works as expected, when using the right header information.
